hippod/app/services/items.js

```diff
@@ -260,16 +260,67 @@ hippoD.factory('AchievementService', function($http) {
+                                                count_in_list = true
+                                                }
+                                            }
+                                        if (!(count_in_list)) {
+                                            if (item_data[i]['name'].includes(data_id)){
+                                                // case of non-given name for snippet image
+                                                entry['name'] = item_data[i]['name'].replace(data_id, counter)
+                                                var obj = {}
+                                                obj[data_id] = counter
+                                                id_memory.push(obj);
+                                                numm++
+                                                }
+                                            // case of snippet image with user given name
+                                            else {entry['name'] = item_data[i]['name']}
+                                        }
+                                        else{entry['name'] = item_data[i]['name'].replace(data_id, same_counter)}
+
                                     } else {
-                                        entry['type'] = 'undefined'
+                                        var count_in_list = false
+                                        for (var k = 0; k < id_memory.length; k++) {
+                                            if (data_id in id_memory[k]) {
+                                                var same_counter = id_memory[k][data_id]
+                                                count_in_list = true
+                                                }
+                                            }
+                                        if (count_in_list){
+                                            entry['type'] = 'undefined';
+                                            entry['name'] = item_data[i]['name'].replace(data_id, same_counter)
+                                            }
+                                        else{
+                                            if (item_data[i]['name'].includes(data_id)){
+                                                entry['name'] = item_data[i]['name'].replace(data_id, counter)
+                                                var obj = {}
+                                                obj[data_id] = counter
+                                                id_memory.push(obj);
+                                                num++
+                                            }
+                                            else{
+                                                entry['type'] = 'undefined';
+                                                entry['name'] = item_data[i]['name'];
+                                                }
+                                            }
```

Comment:
What is here implemented? This looks **way too** complicated. If it is that complicated, it should be fixed somewhere else ... Really